onnx/test/symbolic_shape_test.py

```diff
@@ -0,0 +1,88 @@
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from onnx import checker, helper, TensorProto, NodeProto, GraphProto, ValueInfoProto, ModelProto, ONNX_ML, SparseTensorProto
+from onnx.helper import make_model, make_node, make_tensor, make_tensor_value_info, make_empty_tensor_value_info, make_opsetid, make_tensor_sequence_value_info
+from typing import Sequence, Union, Text, Tuple, List, Any, Optional
+import onnx.shape_inference
+import unittest
+import os
+import numpy as np  # type: ignore
+
+
+class TestSymbolicShape(unittest.TestCase):
+
+    def test_clip_enable_symbolic(self):  # type: () -> None
+
+        clip = helper.make_node('Concat', inputs=['A', 'B'], outputs=['C'], name='Concat', axis=1)
+        cast = onnx.helper.make_node('Cast',
+            inputs=['C'],
+            outputs=['output'],
+            to=getattr(TensorProto, 'FLOAT'))
+        graph_def = helper.make_graph(name='test_graph',
+            nodes=[clip, cast],
+            inputs=[helper.make_tensor_value_info('A', TensorProto.FLOAT, [2, 'A']),
+                helper.make_tensor_value_info('B', TensorProto.FLOAT, [2, 3])],
+            outputs=[helper.make_tensor_value_info('output', TensorProto.FLOAT, [2, 'M'])]
+        )
+
+        onnx_model = make_model(graph_def)
+        inferred_model = onnx.shape_inference.infer_shapes(onnx_model, strict_mode=True)
+        vis = list(inferred_model.graph.value_info)
+        inferred_vis = [make_tensor_value_info("C", TensorProto.FLOAT, (2, 'unk__0'))]
```

Comment:
Instead of checking that the dimension has a specific name, it is better to check that it has some name. (We can do this check for 'C', and/or 'output'.) Alternatively, we can check that 'C' and 'output' both have a name and that the name is the same.

Reply:
Good idea. Instead of checking the exact dim_param name, right now it only checks whether a dim_param is produced. I use "-1" as dim_value to represent a symbolic dim. Also, it further checks whether the symbolic dim of 'C' is same as 'output'.